src/server/json_family.cc

```diff
@@ -1451,39 +1451,42 @@ OpStatus OpMSet(const OpArgs& op_args, const ShardArgs& args) {
 
   return result;
 }
-OpStatus OpMerge(const OpArgs& op_args, string_view key, std::string_view json_str) {
+
+// Note that currently OpMerge works only with jsoncons and json::Path support has not been
+// implemented yet.
+OpStatus OpMerge(const OpArgs& op_args, string_view key, JsonPathV2 expression, string_view path,
+                 std::string_view json_str) {
   std::optional<JsonType> parsed_json = JsonFromString(json_str);
   if (!parsed_json) {
     VLOG(1) << "got invalid JSON string '" << json_str << "' cannot be saved";
     return OpStatus::SYNTAX_ERR;
   }
 
-  auto& db_slice = op_args.GetDbSlice();
-  auto it_res = db_slice.FindMutable(op_args.db_cntx, key, OBJ_JSON);
-  if (it_res.ok()) {
-    op_args.shard->search_indices()->RemoveDoc(key, op_args.db_cntx, it_res->it->second);
+  auto cb = [&](optional<string_view> cur_path, JsonType* val) {
+    string_view strpath = cur_path ? *cur_path : string_view{};
 
-    JsonType* obj = it_res->it->second.GetJson();
+    DVLOG(2) << "Handling " << strpath << " " << val->to_string();
+    // https://datatracker.ietf.org/doc/html/rfc7386#section-2
     try {
-      // https://datatracker.ietf.org/doc/html/rfc7386#section-2
-      mergepatch::apply_merge_patch(*obj, *parsed_json);
+      mergepatch::apply_merge_patch(*val, *parsed_json);
     } catch (const std::exception& e) {
-      LOG_EVERY_T(ERROR, 1) << "Exception in OpMerge: " << e.what() << " with obj: " << *obj
-                            << " and patch: " << *parsed_json;
-      return OpStatus::INVALID_VALUE;
+      LOG_EVERY_T(ERROR, 1) << "Exception in OpMerge: " << e.what() << " with obj: " << *val
+                            << " and patch: " << *parsed_json << ", path: " << strpath;
+      return false;
```

Comment:
Are you sure this is the correct return type? We return `false` here and `false` below. If so then just remove this return and let if fall through ?

From a quick look we call this: 

```
   338   error_code ec = JsonReplace(json_entry, path, callback);
   339   if (ec) {
   340     VLOG(1) << "Failed to evaluate expression on json with error: " << ec.message();
   341     return OpStatus::SYNTAX_ERR;
   342   }
```

so I guess if we `return false` in the callback then we won't return `SyntaxError` ? If I am right, it's worth adding a test as well to keep the behaviour intact for future iterations

Reply:
I am sure it's the correct return type :)
the callback returns true if the current path needs to be deleted.

Reply:
I fully trust you :nerd_face: